compiler/luci-interpreter/src/kernels/TransposeConv.cpp

```diff
@@ -0,0 +1,174 @@
+  op_params.padding_values.height = computePaddingWithOffset(
+      params().stride_height, 1, height, filter_height, unused_output_height, &offset);
+  op_params.padding_values.height_offset = offset;
+  op_params.padding_values.width = computePaddingWithOffset(
+      params().stride_width, 1, width, filter_width, unused_output_width, &offset);
+  op_params.padding_values.width_offset = offset;
+  op_params.stride_height = params().stride_height;
+  op_params.stride_width = params().stride_width;
+  op_params.output_multiplier = _output_multiplier;
+  tflite::reference_ops::TransposeConv(op_params, getTensorShape(_inputData),
+                                       getTensorData<float>(_inputData), getTensorShape(_weights),
+                                       getTensorData<float>(_weights), getTensorShape(_output),
+                                       getTensorData<float>(_output), getTensorShape(_im2col.get()),
+                                       getTensorData<float>(_im2col.get()));
+}
+
+void TransposeConv::evalQuantized() const
+{
+  int32_t input_offset = -_inputData->zero_point();
+  int32_t filter_offset = -_weights->zero_point();
+  int32_t output_offset = _weights->zero_point();
+  const int width = _output->shape().dim(2);
+  const int height = _output->shape().dim(1);
+
+  const int filter_width = _weights->shape().dim(2);
+  const int filter_height = _weights->shape().dim(1);
+
+  int unused_output_height, unused_output_width;
+  unused_output_width =
+      computeOutputSize(params().padding, width, filter_width, params().stride_width, 1);
+  unused_output_height =
+      computeOutputSize(params().padding, height, filter_height, params().stride_height, 1);
+  int32_t offset = 0;
+  tflite::ConvParams op_params;
+  op_params.padding_type = tflite::PaddingType::kSame;
+  op_params.padding_values.height = computePaddingWithOffset(
+      params().stride_height, 1, height, filter_height, unused_output_height, &offset);
+  op_params.padding_values.width = computePaddingWithOffset(
+      params().stride_width, 1, width, filter_width, unused_output_width, &offset);
+  op_params.stride_height = params().stride_height;
+  op_params.stride_width = params().stride_width;
+  op_params.input_offset = input_offset;
+  op_params.output_offset = output_offset;
+  op_params.weights_offset = filter_offset;
+  op_params.output_multiplier = _output_multiplier;
+  op_params.output_shift = -_output_shift;
+  op_params.quantized_activation_min = _output_activation_min;
+  op_params.quantized_activation_max = _output_activation_max;
+
+  tflite::reference_ops::TransposeConv(
+      op_params, getTensorShape(_inputData), getTensorData<uint8>(_inputData),
+      getTensorShape(_weights), getTensorData<uint8>(_weights), getTensorShape(_output),
+      getTensorData<uint8>(_output), getTensorShape(_im2col.get()),
+      getTensorData<uint8>(_im2col.get()), getTensorData<int32_t>(_scratch_tensor.get()));
```

Comment:
AFAIK those temporary tensors are not used in reference implementation, you can just pass nullptr and empty shape.

Reply:
Oh, I found it! I will change to nullptr and empty shape. Thanks a lot!